File: ToolKit/src/main/java/com/uday/android/toolkit/listeners/DeleteClickListener.java
package com.uday.android.toolkit.listeners;

import android.app.*;
import android.content.*;
import android.os.*;
import android.view.*;
import android.view.animation.*;
import android.widget.*;
import com.uday.android.toolkit.*;
import com.uday.android.toolkit.ui.*;
import com.uday.android.util.*;
import eu.chainfire.libsuperuser.*;
import java.util.*;
import android.view.View.*;
import com.uday.android.toolkit.fragments.*;

public class DeleteClickListener implements DialogUtils.OnClickListener{

	private ApkListData apkListData;
	private int position;
	private ApkListAdapter adapter;
	private Context context;
	private Animation anim;
	private Shell.OnCommandResultListener commandResultListener;
	private View row;

	private static final Shell.Interactive rootSession=MainActivity.rootSession;

	public DeleteClickListener(final ApkListAdapter adapter){
		this.adapter=adapter;
		this.context=adapter.getContext();
		anim = AnimationUtils.loadAnimation(
			context, android.R.anim.slide_out_right
		);
		anim.setDuration(500);

		commandResultListener=new Shell.OnCommandResultListener(){
			@Override
			public void onCommandResult(int commandCode,int exitCode,List<String> output){
				((Activity)context).runOnUiThread(new Runnable(){
						@Override
						public void run(){
							if(!apkListData.apkFile.exists()){
								row.startAnimation(anim);
								new Handler().postDelayed(new Runnable(){
										@Override
										public void run(){
											adapter.remove(apkListData);
											BatchInstallerFragment.apkFilesOrig.remove(apkListData);
										}
									},400);
								CustomToast.showSuccessToast(context,apkListData.PATH+" is deleted",Toast.LENGTH_SHORT);
							}
							else
								CustomToast.showFailureToast(context,"failed to delete"+apkListData.PATH,Toast.LENGTH_SHORT);

						}
					});
			}
		};
	}

	@Override
	public void onClick(final AlertDialog sweet)
	{
		sweet.cancel();
		AlertDialog dialog=DialogUtils.showConfirmDialog(context,"Are you sure..?","do you want to permanently delete\n"+apkListData.PATH+"..?",null,"confirm",new DialogUtils.OnClickListener(){
										@Override
										public void onClick(AlertDialog p1){
											rootSession.addCommand(MainActivity.TOOL+" rm -f "+'"'+apkListData.PATH+'"',position,commandResultListener);
											p1.cancel();
										}
									});
		dialog.setIcon(apkListData.ICON);
	}

	public DeleteClickListener setPosition(int position,View row){
		this.position=position;
		this.apkListData=adapter.getItem(position);
		this.row=row;
		return this;
	}
}

File: app/src/main/java/com/uday/android/toolkit/listeners/DeleteClickListener.kt
package com.uday.android.toolkit.listeners

import android.content.Context
import android.os.Handler
import android.view.View
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import com.uday.android.toolkit.MainActivity
import com.uday.android.toolkit.fragments.BatchInstallerFragment
import com.uday.android.toolkit.ui.ApkListAdapter
import com.uday.android.toolkit.ui.CustomToast
import com.uday.android.toolkit.ui.DialogUtils
import com.uday.android.util.ApkListData
import eu.chainfire.libsuperuser.Shell

class DeleteClickListener(private val adapter: ApkListAdapter) : DialogUtils.OnClickListener {

    private var apkListData: ApkListData? = null
    private var position: Int = 0
    private val context: Context = adapter.context
    private val anim: Animation
    private val commandResultListener: Shell.OnCommandResultListener
    private var row: View? = null

    init {
        anim = AnimationUtils.loadAnimation(
            context, android.R.anim.slide_out_right
        )
        anim.duration = 500

        commandResultListener =
            Shell.OnCommandResultListener { commandCode, exitCode, output ->
                (context as AppCompatActivity).runOnUiThread {
                    if (!apkListData!!.apkFile.exists()) {
                        row!!.startAnimation(anim)
                        Handler().postDelayed({
                            adapter.remove(apkListData)
                            BatchInstallerFragment.apkFilesOrig!!.remove(apkListData!!)
                        }, 400)
                        CustomToast.showSuccessToast(
                            context,
                            apkListData!!.PATH + " is deleted",
                            Toast.LENGTH_SHORT
                        )
                    } else
                        CustomToast.showFailureToast(
                            context,
                            "failed to delete" + apkListData!!.PATH,
                            Toast.LENGTH_SHORT
                        )
                }
            }
    }

    override fun onClick(p1: AlertDialog?) {
        p1?.cancel()
        val dialog = DialogUtils.showConfirmDialog(
            context,
            "Are you sure..?",
            "do you want to permanently delete\n" + apkListData!!.PATH + "..?",
            null,
            "confirm",
            object : DialogUtils.OnClickListener {
                override fun onClick(p1: AlertDialog?) {
                    rootSession!!.addCommand(
                        MainActivity.TOOL + " rm -f " + '"'.toString() + apkListData!!.PATH + '"'.toString(),
                        position,
                        commandResultListener
                    )
                    p1!!.cancel()
                }
            })
        dialog.setIcon(apkListData!!.ICON)
    }

    fun setPosition(position: Int, row: View): DeleteClickListener {
        this.position = position
        this.apkListData = adapter.getItem(position)
        this.row = row
        return this
    }

    companion object {

        private val rootSession = MainActivity.rootSession
    }
}
